sycl/doc/MultiTileCardWithLevelZero.md

```diff
@@ -0,0 +1,125 @@
+			   
+`malloc_shared`:
+- Allocation can be accessed by the host and the specified device only.
+- The data can migrate (operated by the Level-Zero driver) between the host and the device for faster access.
+- No explicit copy is necessary for synchronizing between the host and the device, but it is needed for other devices in the context.
+				   
+NOTE: Memory allocated against a root-device is accessible by all of its sub-devices (tiles).
+So if operating on a context with multiple sub-devices of the same root-device then you can use `malloc_device` on that root-device instead of using the slower `malloc_host`.
+Remember that if using `malloc_device` you'd need an explicit copy out to the host if it necessary to see data there.
+					   
+### 3.2 Buffer
+	
+SYCL buffers are also created against a context and under the hood are mapped to the Level-Zero USM allocation discussed above.
+The current mapping is following:
+
+- For integrated device the allocations are made on host, and are accessible by the host and the device without any copying.
+- Memory buffers for context with sub-devices of the same root-device (possibly including the root-device itself) are allocated on that root-device.
+   Thus they are readily accessible by all the devices in such context. The synchronization with the host is performed by SYCL RT with map/unmap doing implicit copies when necessary.
+- Memory buffers for context with devices from different root-devices in it are allocated on host (thus made accessible to all devices).
+	
+## 4 Queue
+
+SYCL queue is always attached to a single device in a possibly multi-device context.
+Some typical scenarios are the following (from most performant to least performant):
+
+A. Context with a single sub-device in it and the queue is attached to that sub-device (tile)
+- The execution/visibility is limited to the single sub-device only
+- Expected to offer the best performance per tile
+		
+B. Context with multiple sub-devices of the same root-device (multi-tile)
+- Queues are to be attached to the sub-devices effectively implementing "explicit scaling"
+- The root-device should not be passed to such context for better performance
+	
+C. Context with a single root-device in it and the queue is attached to that root-device
+- The work will be automatically distributed across all sub-devices/tiles via "implicit scaling" by the driver
+- The most simple way to enable multi-tile HW but doesn't offer possibility to target specific tiles
+		
+D. Contexts with multiple root-devices (multi-card)
+- The most unrestrictive context with queues attached to different root-devices
+- Offers most sharing possibilities at the cost of slow access through host memory or explicit copies needed
+
+Depending on the chosen programming model (A,B,C,D) and algorithm used make sure to do proper memory allocation/synchronization.
+				
+## 5 Examples
+	
+These are few examples of programming to multiple tiles and multiple cards:

```

Comment:
These are external examples which are good, but might be lost over time... can we have something as part of this doc?

Reply:
I wanted to shoe E2E test these would be too large for this doc